src/main/java/WayofTime/alchemicalWizardry/client/nei/NEIMeteorRecipeHandler.java

```diff
@@ -61,10 +65,50 @@ public CachedMeteorRecipe(MeteorParadigm meteor, ItemStack focusStack) {
                     row++;
                 }
 
-                if (focusStack != null && matchItem(focusStack, stack)) {
+                if (matchItem(focusStack, stack)) {
                     this.focus = new Point(xPos - 1, yPos - 1);
                 }
             }
+
+            List<MeteorParadigmComponent> sortedFiller = new ArrayList<>(meteor.fillerList);
+
+            if (fillerChance > 0) {
+                if (col != 0) {
+                    col = 0;
+                    row++;
+                }
+
+                if (!sortedFiller.isEmpty()) {
+                    sortedFiller.sort(Comparator.comparingInt(c -> -c.getChance()));
+                } else {
+                    sortedFiller.add(new MeteorParadigmComponent(new ItemStack(Blocks.stone), 1));
+                    totalFillerWeight = 1;
+                }
+
+                for (MeteorParadigmComponent filler : sortedFiller) {
+                    ItemStack stack = filler.getValidBlockParadigm();
+                    int xPos = 3 + 18 * col;
+                    int yPos = 37 + 18 * row;
+
+                    List<String> tooltips = new ArrayList<>();
+                    float chance = filler.getChance() / totalFillerWeight * (float) (fillerChance / 100.0);
```

Comment:
Precompute `fillerRatio` outside of the loop, as mentioned above.

Reply:
Addressed in [b5f55eb](https://github.com/GTNewHorizons/BloodMagic/pull/60/commits/b5f55eb79eadb8e225dff4fafff22f3db3a52a2f)